### PR TITLE
Move existing code to codegen gradle module

### PR DESCRIPTION
**Pull request by ashwini-desai**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -1,47 +1,50 @@
...
-    id 'application'
+    id 'org.jetbrains.kotlin.jvm' version '1.3.72' apply false
 }
 
-repositories {
-    jcenter()
-    maven { url 'https://jitpack.io' }
-}
+subprojects {
+    apply plugin: 'org.jetbrains.kotlin.jvm'
```
Comment (bhimsenp) [inline build.gradle:6]: Do we need subprojects now (since we have only one module) or this is in case we have runtime module in future?

Comment (ashwini-desai) [reply build.gradle:6]: Introducing ```runtime``` module soon :) Moving exception classes there

File: build.gradle [hunk 2]
```diff
@@ -1,47 +1,50 @@
...
+    }
 
-    implementation 'com.github.medly:ktee:master-SNAPSHOT'
+    dependencies {
+        implementation platform('org.jetbrains.kotlin:kotlin-bom')
+        implementation 'org.jetbrains.kotlin:kotlin-stdlib-jdk8'
 
-    implementation  'org.slf4j:slf4j-simple:1.7.30'
-    implementation 'org.apache.commons:commons-text:1.8'
+        testImplementation 'org.jetbrains.kotlin:kotlin-test'
```
Comment (kdabir) [inline build.gradle:18]: we should move to kotest now.

**Review by kdabir — APPROVED** [review]

**Comment by ashwini-desai** [comment]
@kdabir @bhimsenp I had raised this PR against the refactoring branch since I thought we all would work on that, but I should raise it against master since we would keep on having conflicts otherwise.

**Pull request closed without merging**